MarsRover/Headings/EastHeading.cs

```diff
@@ -0,0 +1,27 @@
+﻿
+namespace MarsRover
+{
+    public class EastHeading : IHeading
+    {
+        public IHeading TurnRight() => new SouthHeading();
+
+        public IHeading TurnLeft() => new NorthHeading();
+
+
+        public Grid MoveForward(Grid grid)
+        {
+            var x = grid.XCoordinate(); 
+            x = (grid.XCoordinate() < grid.GridWidth()) ? x += 1 : 0;
+            return grid = new Grid(x, grid.YCoordinate(), grid.GridWidth(), grid.GridHeight());
+        }
+
+        public Grid MoveBackward(Grid grid)
+        {
+            var x = grid.XCoordinate();
+            x = (grid.XCoordinate() > 0) ? x -= 1 : grid.GridWidth();
+            return grid = new Grid(x, grid.YCoordinate(), grid.GridWidth(), grid.GridHeight());
+        }
+
+        public string GetHeading() => "East";
```

Comment:
`string` can be fragile, it's subject to typos that the compiler can't catch!  Let's put the compiler to work for us and change this to an `enum`.

```cs
public enum Headings
{
    North,
    East,
    South,
    West
}
```

Then you can change this line to:

`public string GetHeading() => Headings.East;`

Now the compiler will work for us, and it makes it IDE friendly too (i.e. autocomplete).

Note, this will need to be applied elsewhere too.

Reply:
I added a class to support the heading properties and those will be called instead of setting a string directly.